modules/programs/hyprpanel/default.nix

```diff
@@ -91,7 +91,26 @@ in
 
     xdg.configFile.hyprpanel = lib.mkIf (cfg.settings != { }) {
       target = "hyprpanel/config.json";
-      source = jsonFormat.generate "hyprpanel-config" cfg.settings;
+      source = jsonFormat.generate "hyprpanel-config" (
+        if cfg.settings ? theme && cfg.settings.theme ? name then
```

Comment:
This is a freeform type using a generator. It doesn't make sense to implement a warning on a deprecation, that never actually was implemented here. It is a warning / documentation that should be provided upstream, for users of their flake and migrating to home-manager.

Reply:
Users who used upstream's module were automatically migrated to this module, so this is a continuation of the old module. It is not backward compatible, but we should at least print a warning

Reply:
Oh that's true. Forgot about the same module option name causing the conflict/migration.